lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -230,9 +245,12 @@ def self.upload_binary(app_id, binary_path)
         end
       end
 
+      # Uploads the binary
+      #
+      # Returns the id of the release"> <explain "Only happens on a successful release
       def self.upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
-        status = upload_status(upload_token, app_id)
+        status, release_id = upload_status(upload_token, app_id)
```

Comment:
If you encapsulate the response in a simple Ruby class as described in my other comment, this could become:

```
upload_status_response = upload_status(upload_token, app_id)

if upload_status_response.success?
   ... do stuff
elsif upload_status_response.in_progress?
   ... do other stuff
else
   ... 
end
```

No worries if you don't want to take this on at this time, but thought it might clean things up in this method a bit.